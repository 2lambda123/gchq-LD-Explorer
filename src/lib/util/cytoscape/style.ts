/* (c) Crown Copyright GCHQ */

export default [
	// the stylesheet for the graph
	{
		selector: 'node',
		style: {
			label: 'data(label)',
			width: 20,
			height: 20
		}
	},
	{
		selector: 'node[termType = "NamedNode"]',
		style: {
			'background-color': '#fbbf24'
		}
	},
	{
		selector: 'node[termType = "Literal"]',
		style: {
			'background-color': '#000'
		}
	},
	{
		selector: 'label',
		style: {
			'font-size': '5px'
		}
	},
	{
		selector: 'edge',
		style: {
			width: 2,
			'line-color': '#C4C8CD',
			'target-arrow-color': '#C4C8CD',
			'target-arrow-shape': 'triangle',
			'curve-style': 'bezier',
			label: 'data(label)'
		}
	}
];
