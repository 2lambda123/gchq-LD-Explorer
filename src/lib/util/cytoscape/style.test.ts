/* (c) Crown Copyright GCHQ */

import style from './style';

describe('style', () => {
	it('matches the snapshot', () => {
		expect(style).toMatchSnapshot();
	});

	it('is a correctly formatted array containing style data', () => {
		const uniqueKeys = Array.from(new Set(style.flatMap(Object.keys)));
		expect(Array.from(uniqueKeys)).toHaveLength(2);
		expect(Array.from(uniqueKeys)).toContain('style');
		expect(Array.from(uniqueKeys)).toContain('selector');
	});
});
