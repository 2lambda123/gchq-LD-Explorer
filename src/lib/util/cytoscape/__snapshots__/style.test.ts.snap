// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`style > matches the snapshot 1`] = `
[
  {
    "selector": "node",
    "style": {
      "height": 20,
      "label": "data(label)",
      "width": 20,
    },
  },
  {
    "selector": "node[termType = "NamedNode"]",
    "style": {
      "background-color": "#fbbf24",
    },
  },
  {
    "selector": "node[termType = "Literal"]",
    "style": {
      "background-color": "#000",
    },
  },
  {
    "selector": "label",
    "style": {
      "font-size": "5px",
    },
  },
  {
    "selector": "edge",
    "style": {
      "curve-style": "bezier",
      "label": "data(label)",
      "line-color": "#C4C8CD",
      "target-arrow-color": "#C4C8CD",
      "target-arrow-shape": "triangle",
      "width": 2,
    },
  },
]
`;
